src/IntegerConversion.hpp

```diff
@@ -106,17 +114,18 @@ template< typename OUTPUT, typename INPUT >
 std::enable_if_t< !internal::canEasilyConvert< INPUT, OUTPUT > &&
                   !std::is_unsigned< INPUT >::value,
                   OUTPUT >
+inline constexpr LVARRAY_HOST_DEVICE
 integerConversion( INPUT input )
 {
   static_assert( std::is_integral< INPUT >::value, "INPUT must be an integral type." );
   static_assert( std::is_integral< OUTPUT >::value, "OUTPUT must be an integral type." );
 
-  LVARRAY_ERROR_IF_LT( input, std::make_signed_t< OUTPUT >{ std::numeric_limits< OUTPUT >::min() } );
+  LVARRAY_ERROR_IF_LT( input, std::make_signed_t< OUTPUT >{ internal::IntegerTraits< OUTPUT >::min } );
```

Comment:
The CUDA compiler on Ascent does not seem to like this `LVARRAY_ERROR_IF_LT` and fails with the error `/ccsopen/home/hamonf/GEOSX/src/coreComponents/LvArray/unitTests/../src/IntegerConversion.hpp(123): error: statement may not appear in a constexpr function`. I get the same error at line 128 of this file, and at line 84 of `StackBuffer.hpp`. When these lines are commented out I can successfully compile so this problem is not blocking.

Reply:
Yeah I didn't mean to make them `constexpr`, nice catch.